airsonic-main/src/test/java/org/airsonic/player/service/MediaScannerServiceTestCase.java

```diff
@@ -105,6 +106,15 @@ public Statement apply(Statement base, Description description) {
     ResourceLoader resourceLoader;
 
 
+    @Before
+    public void setup() {
+        while (mediaScannerService.isScanning()) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {}
+        }
+    }
```

Comment:
This is needed because when Airsonic comes up, the MediaScannerService runs (because it's a "new" test installation). If a test runs a media scan of its own, it is just skipped depending on whether the previous media scan was finished. That's a jiggler waiting to happen